Content.Client/_DV/CartridgeLoader/Cartridges/NanoChatUiFragment.xaml.cs

```diff
@@ -214,18 +230,88 @@ private void UpdateMessages(Dictionary<uint, List<NanoChatMessage>> messages)
             scroll.SetScrollValue(new Vector2(0, float.MaxValue));
     }
 
+    private void UpdateContactList(List<NanoChatRecipient>? contacts)
+    {
+        ContactsList.RemoveAllChildren();
+        if (contacts is null)
+        {
+            ContactsList.AddChild(new Label() { Text = Loc.GetString("nano-chat-look-up-no-server") });
+            return;
+        }
+        for (var idx = 0; idx < contacts.Count; idx++)
+        {
+            var contact = contacts[idx];
+            var nameLabel = new Label()
+            {
+                Text = contact.Name,
+                HorizontalAlignment = HAlignment.Left,
+                HorizontalExpand = true
+            };
+            var numberLabel = new Label()
+            {
+                Text = $"#{contacts[idx].Number:D4}",
+                HorizontalAlignment = HAlignment.Right,
+                Margin = new Thickness(0, 0, 36, 0),
+            };
+            var startChatButton = new Button()
+            {
+                Text = "+",
+                HorizontalAlignment = HAlignment.Right,
+                MinSize = new Vector2(32, 32),
+                MaxSize = new Vector2(32, 32),
+                ToolTip = Loc.GetString("nano-chat-new-chat"),
+            };
+            startChatButton.AddStyleClass("OpenBoth");
+            if (contact.Number == _ownNumber || _recipients.ContainsKey(contact.Number))
+            {
+                startChatButton.Disabled = true;
+            }
+            startChatButton.OnPressed += _ =>
+            {
+                if (OnMessageSent is { } handler)
+                {
+                    handler(NanoChatUiMessageType.NewChat, contact.Number, contact.Name, contact.JobTitle);
+                    SelectChat(contact.Number);
+                    ToggleView();
+                }
+            };
+
+            var panel = new PanelContainer()
+            {
+                HorizontalExpand = true,
+            };
+
+            panel.AddChild(nameLabel);
+            panel.AddChild(numberLabel);
+            panel.AddChild(startChatButton);
```

Comment:
move this into a separate control using xaml

Reply:
![](https://github.com/user-attachments/assets/2da629c6-e427-4dcd-8931-48fedcb4e5bc)
Can you confirm/deny that this is approximately what you expected?
The normal view still in the same place, and just the LookupView separated into its own Container?